src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -56,24 +56,26 @@ public bool TryReadTo(out ReadOnlySpan<T> span, T delimiter, T delimiterEscape,
             ReadOnlySpan<T> remaining = UnreadSpan;
             int index = remaining.IndexOf(delimiter);
 
-            if (index != -1)
+            if (index < 0)
             {
-                if (index > 0 && remaining[index - 1].Equals(delimiterEscape))
-                {
-                    // This delimiter might be skipped, go down the slow path
-                    return TryReadToSlow(out span, delimiter, delimiterEscape, advancePastDelimiter);
-                }
-                span = index == 0 ? default : remaining.Slice(0, index);
+                goto SlowPath;
+            }
+
+            if (index == 0 || !remaining[index - 1].Equals(delimiterEscape))
```

Comment:
Wouldn't it be clearer to combine the if's here?

``` C#
if (index == 0 || (index > 0 && !remaining[index - 1].Equals(delimiterEscape)))
```
```

Reply:
Yes, that would work nicely and avoid the multiple return statements.

The goto was there to avoid multiple return points.

Reply:
nit: I would invert the order of the checks though, since index > 0 would be the more common case

``` C#
if ((index > 0 && !remaining[index - 1].Equals(delimiterEscape)) || index == 0)
```

Reply:
Sure, can you follow up?

Reply:
Yep, will do.

Reply:
Done: https://github.com/dotnet/corefxlab/pull/2514